src/apps/loan-list/modal/renew-loans-modal-content.tsx

```diff
@@ -0,0 +1,133 @@
+import React, { useState, useEffect, FC } from "react";
+import CheckBox from "../materials/utils/checkbox";
+import { LoanV2 } from "../../../core/fbs/model/loanV2";
+import SelectableMaterial from "../materials/selectable-material";
+import { useRenewLoansV2 } from "../../../core/fbs/fbs";
+import { RenewedLoanV2 } from "../../../core/fbs/model";
+import { getRenewableMaterials, getAmountOfRenewableLoans } from "../helpers";
+
+interface RenewLoansModalContentProps {
+  renewable: number | null;
+  loansModal: LoanV2[];
+  buttonLabel: string;
+  checkboxLabel: string;
+}
+
+const RenewLoansModalContent: FC<RenewLoansModalContentProps> = ({
+  renewable,
+  loansModal,
+  checkboxLabel,
+  buttonLabel
+}) => {
+  const { mutate } = useRenewLoansV2();
+  const [materialsToRenew, setMaterialsToRenew] = useState<number[]>([]);
+  const [allRenewableMaterials, setAllRenewableMaterials] = useState<number>(0);
+  const [loans, setLoans] = useState<Array<LoanV2>>([]);
+  const [renewedLoans, setRenewedLoans] = useState<
+    Array<RenewedLoanV2> | undefined | null
+  >(null);
+
+  const renewSelected = () => {
+    mutate(
+      {
+        data: materialsToRenew
+      },
+      {
+        onSuccess: (result) => {
+          if (result) {
+            const renewedIds = result.map(
+              ({ loanDetails }) => loanDetails.recordId
+            );
+            const filteredLoans = loans.filter((item) => {
+              return renewedIds.indexOf(item.loanDetails.recordId) === -1;
+            });
+            setMaterialsToRenew([]);
+            setLoans(filteredLoans);
+            setRenewedLoans(result);
+          }
+        },
+        // todo error handling, missing in figma
+        onError: () => {}
```

Comment:
I agree on this though it is not satisfying. We do not have any specs og design describing error handling

Reply:
I've flagged it and they will return with something